community/cypher/expressions-3.4/src/main/scala/org/neo4j/cypher/internal/v3_4/expressions/Expression.scala

```diff
@@ -53,6 +73,12 @@ abstract class Expression extends ASTNode {
       acc => (acc :+ e, None)
   }
 
+  // Collects all sub-expressions recursively
+  def subExpressions: Seq[Expression] = this.treeFold(List.empty[Expression]) {
+    case e: Expression if e != this =>
+      acc => (acc :+ e, Some(identity))
```

Comment:
True. It is copied from `arguments()` above it so I guess it applies to that too then. I am not sure if the order matters here, otherwise it could just add to head.